### PR TITLE
GFORMS-1743: Add date&time submission field in robotics xml

### PR DESCRIPTION
**Pull request by lxol**

### CODE REVIEW

File: test/uk/gov/hmrc/gform/submission/RoboticsXMLGeneratorSpec.scala [hunk 1]
```diff
@@ -128,7 +128,7 @@ case object RoboticsXMLGeneratorSpec extends Spec {
     val expected: Elem =
       <gform id={formId.value} dms-id={dmsId} submission-reference={submissionRef.value}>{
         removeWhitespace(expectedFields)
-      }<dateSubmitted>02/01/2019</dateSubmitted></gform>
+      }<dateSubmitted>02/01/2019</dateSubmitted><datetimeSubmitted>2019-01-02T00:00:00Z</datetimeSubmitted></gform>
```
Comment (josef-vlach) [inline test/uk/gov/hmrc/gform/submission/RoboticsXMLGeneratorSpec.scala:131]: Can we cover BST date also?

Comment (lxol) [reply test/uk/gov/hmrc/gform/submission/RoboticsXMLGeneratorSpec.scala:131]: updated the test, The instant time is 1 hours earlier than local BST time

File: test/uk/gov/hmrc/gform/submission/RoboticsXMLGeneratorSpec.scala [hunk 2]
```diff
@@ -128,12 +128,21 @@ case object RoboticsXMLGeneratorSpec extends Spec {
...
 
     val dateSubmitted = Instant.from(LocalDate.of(2019, 1, 2).atStartOfDay(ZoneId.of("Europe/London")))
 
+    val dateSubmittedBST = Instant.from(LocalDate.of(2019, 6, 1).atStartOfDay(ZoneId.of("Europe/London")))
+
+    val expectedBST: Elem =
+      <gform id={formId.value} dms-id={dmsId} submission-reference={submissionRef.value}>{
+        removeWhitespace(expectedFields)
+      }<dateSubmitted>01/06/2019</dateSubmitted><datetimeSubmitted>2019-05-31T23:00:00Z</datetimeSubmitted></gform>
```
Comment (josef-vlach) [inline test/uk/gov/hmrc/gform/submission/RoboticsXMLGeneratorSpec.scala:140]: I don't think this look good for casual reader.

```
01/06/2019
2019-05-31T23:00:00Z
```
One day is in May another day is in June. I think we should display `datetimeSubmitted` with zone information ie `+01:00`

Comment (lxol) [reply test/uk/gov/hmrc/gform/submission/RoboticsXMLGeneratorSpec.scala]: I replaced `ISO_INSTANT` to `ISO_DATE_TIME` that does have zone information

File: test/uk/gov/hmrc/gform/submission/RoboticsXMLGeneratorSpec.scala [hunk 3]
```diff
@@ -128,12 +128,21 @@ case object RoboticsXMLGeneratorSpec extends Spec {
...
+      }<dateSubmitted>02/01/2019</dateSubmitted><datetimeSubmitted>2019-01-02T00:00:00Z[Europe/London]</datetimeSubmitted></gform>
 
     val dateSubmitted = Instant.from(LocalDate.of(2019, 1, 2).atStartOfDay(ZoneId.of("Europe/London")))
 
+    val dateSubmittedBST = Instant.from(LocalDate.of(2019, 6, 1).atStartOfDay(ZoneId.of("Europe/London")))
+
+    val expectedBST: Elem =
+      <gform id={formId.value} dms-id={dmsId} submission-reference={submissionRef.value}>{
+        removeWhitespace(expectedFields)
+      }<dateSubmitted>01/06/2019</dateSubmitted><datetimeSubmitted>2019-06-01T00:00:00+01:00[Europe/London]</datetimeSubmitted></gform>
```
Comment (josef-vlach) [inline test/uk/gov/hmrc/gform/submission/RoboticsXMLGeneratorSpec.scala:140]: Hm, this is better but I've never seen `[Europe/London]` part in practice. Can we get formatting without that? Or rather let's discuss this with the team, if it is ok like that.

File: app/uk/gov/hmrc/gform/submission/RoboticsXMLGenerator.scala [hunk 4]
```diff
@@ -35,13 +37,23 @@ object RoboticsXMLGenerator {
...
 
   private def dateSubmitted(now: Instant): NodeSeq =
     <dateSubmitted>{dateSubmittedFormater.format(now)}</dateSubmitted>
 
+  private val datetimeSubmittedFormater = new DateTimeFormatterBuilder()
+    .append(ISO_LOCAL_DATE_TIME)
+    .optionalStart()
+    .appendOffsetId()
+    .optionalStart()
+    .toFormatter()
```
Comment (josef-vlach) [inline app/uk/gov/hmrc/gform/submission/RoboticsXMLGenerator.scala:52]: It seems there is predefined formatter:
```suggestion
  private val datetimeSubmittedFormater = DateTimeFormatter.ISO_OFFSET_DATE_TIME
```

Comment (lxol) [reply app/uk/gov/hmrc/gform/submission/RoboticsXMLGenerator.scala]: done, thanks!

**Pull request merged**